sklearn/preprocessing/_encoders.py

```diff
@@ -673,37 +685,69 @@ def _legacy_transform(self, X):
 
         return out if self.sparse else out.toarray()
 
+    def _make_onehot_sparse_matrix(self, labels, mask, cat_ns):
+        flat_mask = mask.ravel()
+        n_values = np.array([0] + cat_ns)
+        feature_indices = np.cumsum(n_values)
+
+        indices = (labels + feature_indices[:-1]).ravel()[flat_mask]
+        indptr = mask.sum(axis=1).cumsum()
+        indptr = np.insert(indptr, 0, 0)
+        data = np.ones(len(indices))
+
+        out = sparse.csr_matrix((data, indices, indptr),
+                                shape=(len(labels), feature_indices[-1]),
+                                dtype=self.dtype)
+        return out
+
+    def _make_nan_sparse_matrix(self, Xi_missing, n_categories):
```

Comment:
please document what these parameters are and the purpose/return of this function